### PR TITLE
[MCI] Rest and status layer changes

### PR DESCRIPTION
**Pull request by swathinsankaran**
This PR contains the following changes:
1. Rest layer changes to identify multi-cluster ingress VSes
2. Status layer changes to update annotations and status
3. Ingestion, Graph and rest layer UTs

### CODE REVIEW

**Comment by swathinsankaran** [comment]
build ako

**Review by monotosh-avi — APPROVED** [review]
Looks good to me, though we may have to add changes in SyncObjectStatuses() to support statuc sync during bootup, cc @chauhanshubham

File: internal/status/multicluster_ing_status.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+	akov1alpha1 "github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/pkg/apis/ako/v1alpha1"
+	"github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/pkg/utils"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// DeleteMultiClusterIngressStatusAndAnnotation is a wrapper function which gets the Multi-cluster ingress object and deletes the
+// status/annotations.
+func DeleteMultiClusterIngressStatusAndAnnotation(key string, option *UpdateOptions) {
```
Comment (RiyaGaniga-avi) [inline internal/status/multicluster_ing_status.go:33]: The function is named `DeleteMultiClusterIngressStatusAndAnnotation` but I don't see any annotations being delete.

File: internal/status/multicluster_ing_status.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+			return
+		}
+		UpdateMultiClusterIngressStatus(key, updatedMCI, status, retry+1)
+		return
+	}
+
+	utils.AviLog.Infof("key: %s, msg: Successfully updated the multicluster ingress %s/%s status %+v", key, mci.Namespace, mci.Name, utils.Stringify(status))
+}
+
+func patchMultiClusterIngressAnnotations(mci *akov1alpha1.MultiClusterIngress, vsAnnotations map[string]string, key string) error {
```
Comment (RiyaGaniga-avi) [inline internal/status/multicluster_ing_status.go:105]: Can we rename this to `UpdateMultiClusterIngressAnnotations`? Asking this just for consistency between `UpdateMultiClusterIngressStatus` and `UpdateMultiClusterIngressAnnotations`.

**Comment by chauhanshubham** [comment]
Let's push in the status updates during full sync post reboot in a subsequent PR. @swathinsankaran

**Review by chauhanshubham — APPROVED** [review]

File: internal/status/multicluster_ing_status.go (hunk 1 above)
Comment (swathinsankaran) [reply internal/status/multicluster_ing_status.go:33]: sorry, I missed adding the code.

File: internal/status/multicluster_ing_status.go (hunk 2 above)
Comment (swathinsankaran) [reply internal/status/multicluster_ing_status.go]: I have made the changes.

**Review by RiyaGaniga-avi — APPROVED** [review]

**Comment by swathinsankaran** [comment]
build ako

**Comment by swathinsankaran** [comment]
build ako

**Comment by swathinsankaran** [comment]
build ako

**Review by monotosh-avi — APPROVED** [review]

**Pull request merged**